images/virtualization-artifact/pkg/controller/vi/vi_webhook.go

```diff
@@ -64,8 +65,14 @@ func (v *Validator) ValidateUpdate(_ context.Context, oldObj, newObj runtime.Obj
 	}
 
 	ready, _ := service.GetCondition(vicondition.ReadyType, newVI.Status.Conditions)
-	if newVI.Status.Phase == virtv2.ImageReady || ready.Status == metav1.ConditionTrue {
-		return nil, fmt.Errorf("VirtualImage is in the Ready state: spec is immutable now")
+	if ready.Status == metav1.ConditionTrue || newVI.Status.Phase == virtv2.ImageReady || newVI.Status.Phase == virtv2.ImageLost || newVI.Status.Phase == virtv2.ImageTerminating {
+		if !reflect.DeepEqual(oldVI.Spec.DataSource, newVI.Spec.DataSource) {
+			return nil, fmt.Errorf("VirtualImage has already been created: data source cannot be changed after disk is created")
+		}
+
+		if !reflect.DeepEqual(oldVI.Spec.PersistentVolumeClaim.StorageClass, newVI.Spec.PersistentVolumeClaim.StorageClass) {
+			return nil, fmt.Errorf("VirtualImage has already been created: storage class cannot be changed after disk is created")
```

Comment:
Can we allow to change storage class to actual storage class that was used in the PVC?

Reply:
We allow changing the `StorageClass` for a `VirtualImage` in the `Not Ready` state, just as we do for a `VirtualDisk`.